provider_mock_test.go

```diff
@@ -102,40 +116,48 @@ func (m *mockProvider) CreateServicePrincipal(_ context.Context, _ string, _ tim
 	return id, pass, nil
 }
 
-func (m *mockProvider) CreateApplication(_ context.Context, _ string) (api.ApplicationResult, error) {
+func (m *mockProvider) CreateApplication(_ context.Context, _ string) (models.Applicationable, error) {
+	if m.ctxTimeout != 0 {
```

Comment:
I added a time.Sleep to simulate the context deadline error expected by [`TestSP_WAL_Cleanup`  ](https://github.com/hashicorp/vault-plugin-secrets-azure/pull/169/files#diff-77464a25e41d440bcffea845c2842cd0c3d57deafa53c2391826d780b7bde89aR96)

Reply:
If we keep this approach, can we make `ctxTimeout` a `time.Duration` and remove the hardcoded `time.Second` below?

Reply:
Sure thing, I can update that 👍🏼

Reply:
Updated!